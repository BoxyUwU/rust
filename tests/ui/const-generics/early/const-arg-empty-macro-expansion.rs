macro_rules! empty {
    () => {};
}

struct Foo<const N: usize>;

#[rustfmt::skip]
impl Foo<{ empty!{}; }> {}
//~^ ERROR: mismatched types
#[rustfmt::skip]
impl Foo<{ empty!(); }> {}
//~^ ERROR: mismatched types

fn main() {}
