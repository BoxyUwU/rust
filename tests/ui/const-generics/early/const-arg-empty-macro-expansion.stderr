error[E0308]: mismatched types
  --> $DIR/const-arg-empty-macro-expansion.rs:8:10
   |
LL | impl Foo<{ empty!{}; }> {}
   |          ^^^^^^^^^^^^^ expected `usize`, found `()`

error[E0308]: mismatched types
  --> $DIR/const-arg-empty-macro-expansion.rs:11:10
   |
LL | impl Foo<{ empty!(); }> {}
   |          ^^^^^^^^^^^^^ expected `usize`, found `()`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
